server/backend/opendistro_security_client.ts

```diff
@@ -0,0 +1,55 @@
+import { IClusterClient, KibanaRequest } from "../../../../src/core/server";
+import { User} from "../auth/user";
+
+export class SecurityClient {
+  constructor(private readonly esClient: IClusterClient) {
+  }
+
+  public async authenticate(request: KibanaRequest, credentials: any): Promise<User> {
+    const authHeader = Buffer.from(`${credentials.username}:${credentials.password}`).toString('base64');
+    try {
+      let esResponse = await this.esClient.asScoped(request).callAsCurrentUser('opendistro_security.authinfo', {
+        headers: {
+          authorization: `Basic ${authHeader}`,
+        }
+      });
+      return new User(credentials.username, esResponse.roles, esResponse.backend_roles, esResponse.teanats, esResponse.user_requested_tenant, credentials, credentials);
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  }
+
+  public async authenticateWithHeader(request: KibanaRequest, headerName: string, headerValue: string, whitelistedHeadersAndValues: any, additionalAuthHeaders: any = {}): Promise<User> {
+    try {
+      const credentials: any = {
+        headerName,
+        headerValue,
+      };
+      let headers = {};
+      if (headerValue) {
+        headers[headerName] = headerValue;
+      }
+
+      // cannot get config elasticsearch.requestHeadersWhitelist from kibana.yml file in new platfrom
+      // meanwhile, do we really need to save all headers in cookie?
```

Comment:
i recall, there is design at backend plugin using this to determine request coming from kibana node to elasticsearch. in that case bypass auth check at backend if request is from kibana

Reply:
Here is the story of this comment:

the logic in the existing security plugin is that, for all HTTP headers that are in `elasticsearch. requestHeadersWhitelist ` config, and save them in the cookie. then when next time browser makes request with this cookie, kibana server can extract these headers from cookie and add them to the request to Elasticsearch.

while in the new Kibana plugin platform, configurations are scoped by plugins, e.g. security plugin can only get the configurations starts with `opendistro_security.`. So I left a comment here and didn't implement the logic to parse all headers.

based on this. I think maybe it is not necessary for us to save all headers in the cookie, leave them in the header  isn't bad at all.